torchgeo/datasets/ucmerced.py

```diff
@@ -0,0 +1,212 @@
+        "baseballdiamond",
+        "beach",
+        "buildings",
+        "chaparral",
+        "denseresidential",
+        "forest",
+        "freeway",
+        "golfcourse",
+        "harbor",
+        "intersection",
+        "mediumresidential",
+        "mobilehomepark",
+        "overpass",
+        "parkinglot",
+        "river",
+        "runway",
+        "sparseresidential",
+        "storagetanks",
+        "tenniscourt",
+    ]
+
+    class_counts = {class_name: 100 for class_name in classes}
+    class_name_to_label_idx = {class_name: i for i, class_name in enumerate(classes)}
+
+    def __init__(
+        self,
+        root: str = "data",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new UC Merced dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            RuntimeError: if ``download=False`` and data is not found, or checksums
+                don't match
+        """
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.fns = []
+        self.labels = []
+        for class_name, class_count in self.class_counts.items():
+            for i in range(0, class_count):
+                self.fns.append(
+                    os.path.join(
+                        self.root, self.base_dir, class_name, f"{class_name}{i:02d}.tif"
+                    )
+                )
+                self.labels.append(self.class_name_to_label_idx[class_name])
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        sample: Dict[str, Tensor] = {
+            "image": self._load_image(index),
+            "label": torch.tensor(self.labels[index]),  # type: ignore[attr-defined]
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.labels)
+
+    @lru_cache()
```

Comment:
No reason to cache this, the cache will almost never be hit since it only keeps the 128 most recent images and the dataset size is 2.1K.